app/utils/IncomeSourcesUtils.scala

```diff
@@ -42,14 +42,14 @@ trait IncomeSourcesUtils extends FeatureSwitching {
     }
   }
 
-  def newWithIncomeSourcesRemovedFromSession(redirect: Result, sessionService: SessionService, errorRedirect: Result)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+  def withIncomeSourcesRemovedFromSession(redirect: Result)(implicit user: MtdItUser[_], sessionService: SessionService, ec: ExecutionContext): Future[Result] = {
     sessionService.remove(SessionKeys.incomeSourcesSessionKeys, redirect).map {
-      case Left(_) => errorRedirect
+      case Left(_) => throw new Exception("Failed to remove income sources from session")
       case Right(result) => result
     }
   }
 
-  def withIncomeSourcesRemovedFromSession(redirect: Result)(implicit user: MtdItUser[_]): Result = {
+  def withIncomeSourcesRemovedFromSessionLegacy(redirect: Result)(implicit user: MtdItUser[_]): Result = {
     val incomeSourcesSessionKeys = Seq(
       "addUkPropertyStartDate",
```

Comment:
I understand that this is not your code, but we should use refer to variables defined in the SessionKeys, as this is a duplicate definition; pls update this method accordingly as in the scope of your changes

Reply:
There is a method already defined under SessionService which deals with removing session key/value pairs;
if we can re-use existing one and drop use of this method